src/starknetCli.ts

```diff
@@ -164,75 +257,103 @@ export async function runStarknetDeploy(filePath: string, options: IDeployProps)
     let classHash;
     if (!options.no_wallet) {
       assert(compileResult.resultPath !== undefined);
-      classHash = declareContract(compileResult.resultPath, options.network);
+      classHash = declareContract(compileResult.resultPath, false, options);
     }
     const classHashOption = classHash ? `--class_hash ${classHash}` : '';
-    const resultPath = compileResult.resultPath;
-    execSync(
-      `${warpVenvPrefix} starknet deploy --network ${options.network} ${
-        options.no_wallet
-          ? `--no_wallet --contract ${resultPath} `
-          : options.wallet === undefined
-          ? `${classHashOption}`
-          : `${classHashOption} --wallet ${options.wallet}`
-      } ${inputs} ${options.account !== undefined ? `--account ${options.account}` : ''}`,
-      {
-        stdio: 'inherit',
-      },
-    );
-  } catch {
-    logError('starknet deploy failed');
+    const resultPath = compileResult.resultPath as string;
+    const cmd: string = starkNetDeploy(resultPath, options, inputs, classHashOption);
+    callCairoCommand(cmd, CAIRO_CMD_STARKNET_DEPLOY, true, isTest);
+
+    return cmd;
+  } catch (e) {
+    logCLIError('starknet deploy failed');
   }
 }
 
-export function runStarknetDeployAccount(options: IDeployAccountProps) {
+export function deployAccount(option: IDeployAccountProps): string {
+  const account: string = option?.account ? `--account ${option.account}` : '';
+
+  const options: Map<string, string | undefined> = new Map([
+    ['wallet', option?.wallet],
+    ['network', option?.network],
+  ]);
+
+  const multiOptions: string[] = [account];
+
+  const cmd: string = buildCairoCommand(options, CAIRO_CMD_STARKNET_DEPLOY_ACC, multiOptions);
+
+  return cmd;
+}
+
+export function runStarknetDeployAccount(options: IDeployAccountProps, isTest = false) {
   if (options.wallet == undefined) {
-    logError(
+    logCLIError(
       `Error: AssertionError: --wallet must be specified with the "deploy_account" subcommand.`,
     );
     return;
   }
   if (options.network == undefined) {
-    logError(
+    logCLIError(
       `Error: Exception: feeder_gateway_url must be specified with the "deploy_account" subcommand.\nConsider passing --network or setting the STARKNET_NETWORK environment variable.`,
     );
     return;
   }
 
-  const account = options.account ? `--account ${options.account}` : '';
+  const cmd: string = deployAccount(options);
+  callCairoCommand(cmd, CAIRO_CMD_STARKNET_DEPLOY_ACC, true, isTest);
 
-  try {
-    execSync(
-      `${warpVenvPrefix} starknet deploy_account --wallet ${options.wallet} --network ${options.network} ${account}`,
-      {
-        stdio: 'inherit',
-      },
-    );
-  } catch {
-    logError('starknet deploy failed');
-  }
+  return cmd;
+}
+
+export function starknetCallOrInvoke(
```

Comment:
Could you add a suffix here like starknetCallOrInvokeCmd to show that this is where the command is being produced.

Reply:
ok changes made